src/editor/action/mouse/mouse_action_map_label.cpp

```diff
@@ -108,6 +90,47 @@ editor_action* mouse_action_map_label::up_right(editor_display& disp, int x, int
 	return new editor_action_label_delete(hex);
 }
 
+editor_action* mouse_action_map_label::drag_end_left(editor_display& disp, int x, int y)
+{
+	if (click_) return nullptr;
+
+	map_location hex = disp.hex_clicked_on(x, y);
+	if (!disp.get_map().on_board(hex))
+		return nullptr;
+
+	const terrain_label* dragged_label = disp
+		.get_controller()
+		.get_current_map_context()
+		.get_labels()
+		.get_label(clicked_on_);
+
+	// Return nullptr if the dragged label doesn't exist
+	if (!dragged_label) {
+		return nullptr;
+	}
+
+	// Create action chain to:
+	// 1. Delete label in initial hex
+	// 2. Delete label in target hex
```

Comment:
Please say something on this line like "... if it exists, will be a no-op if it doesn't"